internal/envoy/route.go

```diff
@@ -33,10 +33,10 @@ func Routes(routes ...*envoy_api_v2_route.Route) []*envoy_api_v2_route.Route {
 }
 
 // Route returns a *envoy_api_v2_route.Route for the supplied match and action.
-func Route(match *envoy_api_v2_route.RouteMatch, action *envoy_api_v2_route.Route_Route) *envoy_api_v2_route.Route {
+func Route(route *dag.Route) *envoy_api_v2_route.Route {
```

Comment:
I think that this change is a bit awkward because the API signature implies that for any kind of `dag.Route`, the correct `envoy_api_v2_route.Route` form will be returned, but that's not what happens. I'd prefer a change that names the API more specifically.

Reply:
If I'm not mistaken the HTTP -> HTTPS upgrade path is the main exception to this, or are you thinking about something else?

Reply:
Do you have a name in mind?

Reply:
I thought about this more and I think that for now we should just remove both `envoy.Route` and `envoy.Routes`. The latter isn't used, and I'd argue that the former doesn't really provide much value in its current form.

I'm happy to remove it now, and see whether we want to bring in back in a different form later.

Reply:
ok, these are gone :)